filesystem/azure.go

```diff
@@ -0,0 +1,196 @@
+package filesystem
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"github.com/MSOpenTech/azure-sdk-for-go/storage"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type AzureClient struct {
+	client     *storage.StorageClient
+	blobClient *storage.BlobStorageClient
+}
+
+type AzureFile struct {
+	path   string
+	logger *log.Logger
+	client *storage.BlobStorageClient
+}
+
+// convertToAzurePath function
```

Comment:
please make the comment more clear or simply remove the comment for non-public function.

convertToAzurePath splits the given name into two parts. 
The first part ...
It will return any error while converting.